src/NServiceBus.AcceptanceTests/Config/When_startup_is_complete.cs

```diff
@@ -12,48 +13,58 @@ public class When_startup_is_complete : NServiceBusAcceptanceTest
         public async Task Settings_should_be_available_via_DI()
         {
             var context = await Scenario.Define<Context>()
-                    .WithEndpoint<StartedEndpoint>()
-                    .Done(c => c.IsDone)
-                    .Run();
+                .WithEndpoint<StartedEndpoint>()
+                .Done(c => c.IsDone)
+                .Run();
 
             Assert.True(context.SettingIsAvailable, "Setting should be available in DI");
         }
 
-        public class Context : ScenarioContext
+        class Context : ScenarioContext
         {
             public bool IsDone { get; set; }
             public bool SettingIsAvailable { get; set; }
         }
 
-        public class StartedEndpoint : EndpointConfigurationBuilder
+        class AfterConfigIsCompleteFeature : Feature
         {
-            public StartedEndpoint()
+            protected override void Setup(FeatureConfigurationContext context)
             {
-                EndpointSetup<DefaultServer>();
+                context.RegisterStartupTask(b => new AfterConfigIsCompleteFeatureTask(context.Settings, b.Build<Context>()));
```

Comment:
The test is named `Settings_should_be_available_via_DI`. So it doesn't seem correct to manually provide a factory method effectively skipping DI for the settings.